scanner/src/main/kotlin/scanners/fossid/FossIdScanResults.kt

```diff
@@ -196,3 +204,37 @@ private fun urlToPackageType(url: String): PurlType =
             }
         }
     }
+
+/**
+ * FossID returns the matched lines as such:
+ *          255,256,258
+ * This function replaces the map content with:
+ *          255-256, 258
+ * and return it as [TextLocation]s.
+ */
+internal fun List<Int>.compressMatchedLines(): List<TextLocation> {
```

Comment:
I'd prefer if we could instead slightly generalize the `CopyrightStatementsProcessor`'s `getYearRanges()` function by renaming it (to e.g. "collapseRanges"), moving it to `Utils.kt`, and them making use of it here.

Reply:
Done. Please have a look how I adapted the wording of the documentation.